cli/src/cmd/utils.rs

```diff
@@ -381,10 +401,19 @@ impl ScriptSequence {
                                 name: artifact.name.clone(),
                             };
 
+                            // We strip the build metadadata information, since it can lead to
+                            // etherscan not identifying it correctly. eg:
+                            // `v0.8.10+commit.fc410830.Linux.gcc` != `v0.8.10+commit.fc410830`
+                            let version = Version::new(
+                                artifact.version.major,
+                                artifact.version.minor,
+                                artifact.version.patch,
+                            );
```

Comment:
I believe the commit is required: https://etherscan.io/solcversions

what we should do instead is strip the arch and compiler metadata if possible

Reply:
It is, but `forge verify` fills in the rest with `ensure_solc_build_metadata`